src/thor/costmatrix.cc

```diff
@@ -41,13 +41,13 @@ inline float find_percent_along(const valhalla::Location& location, const GraphI
 namespace valhalla {
 namespace thor {
 
-class CostMatrix::TargetMap : public robin_hood::unordered_map<uint64_t, std::vector<uint32_t>> {};
+class CostMatrix::ReachedMap : public robin_hood::unordered_map<uint64_t, std::vector<uint32_t>> {};
 
 // Constructor with cost threshold.
 CostMatrix::CostMatrix(const boost::property_tree::ptree& config)
     : mode_(travel_mode_t::kDrive), access_mode_(kAutoAccess), source_count_(0),
       remaining_sources_(0), target_count_(0), remaining_targets_(0),
-      current_cost_threshold_(0), targets_{new TargetMap},
+      current_cost_threshold_(0), targets_{new ReachedMap}, sources_{new ReachedMap},
```

Comment:
I didn't see another way than keeping the same huge data structure for `sources` as we have for `targets`. bit annoying but as you'll see in the tests (and by fixing the initial issue), maybe it's worth it